src/app/georeferencer/qgsgeorefdatapoint.cpp

```diff
@@ -16,17 +16,21 @@
 
 #include "qgsmapcanvas.h"
 #include "qgsgcpcanvasitem.h"
+#include "qgscoordinatereferencesystem.h"
+
 
 #include "qgsgeorefdatapoint.h"
 
 QgsGeorefDataPoint::QgsGeorefDataPoint( QgsMapCanvas *srcCanvas, QgsMapCanvas *dstCanvas,
                                         const QgsPointXY &pixelCoords, const QgsPointXY &mapCoords,
-                                        bool enable )
+                                        const QString &wktProj, bool enable )
   : mSrcCanvas( srcCanvas )
   , mDstCanvas( dstCanvas )
   , mPixelCoords( pixelCoords )
   , mMapCoords( mapCoords )
+  , mTransCoords( mapCoords )
   , mId( -1 )
+  , mProj( wktProj )
```

Comment:
That just seems to be a recipe for disaster. I'd either create a QgsCoordinateReferenceSystem in the constructor, or even better, instead of passing on a string reference as function parameter, pass on a QgsCoordinateReferenceSystem.

Also, rename the variable to mCrs to ease readability.

Reply:
changed